sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Security/Ucfg/MethodId.cs

```diff
@@ -0,0 +1,63 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2018 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+
+namespace SonarAnalyzer.Security.Ucfg
+{
+    public static class MethodId
+    {
+        public static string Get(ISymbol symbol)
+        {
+            switch (symbol.Kind)
+            {
+                case SymbolKind.Method:
+                    var method = ((IMethodSymbol)symbol).OriginalDefinition;
+                    return $"{Get(method.ContainingType)}.{method.Name}{TypeArguments(method)}({Join(method.Parameters.Select(Get))})";
+                case SymbolKind.Parameter:
+                    var p = (IParameterSymbol)symbol;
+                    return $"{Get(p.Type)}";
+                case SymbolKind.NamedType:
+                    var namedType = (INamedTypeSymbol)symbol;
```

Comment:
What about nested types?

Reply:
There is a test with nested type, it works.